staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -335,9 +336,8 @@ func NewConfig(codecs serializer.CodecFactory) *Config {
 			klog.Fatalf("error getting hostname for apiserver identity: %v", err)
 		}
 
-		h := fnv.New32a()
-		h.Write([]byte(hostname))
-		id = "kube-apiserver-" + fmt.Sprint(h.Sum32())
+		hash := sha256.Sum256([]byte(hostname))
+		id = "kube-apiserver-" + hex.EncodeToString(hash[:])
```

Comment:
This will be 64 characters long, I'd suggest maybe `hash[:16]` to get it down to 32 characters. Or use encoding/base32 or 64 (but might need to remove forbidden characters). I thought we had a base58 encoding package in use but I can't seem to find it if so.

Reply:
I still do not understand why we feel the need to truncate here.  Are we expecting people to interact with these values somehow?  That being said normally I use `base32`:

```go
"kube-apiserver-" + strings.ToLower(base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(hash[:]))
```

which gives you something like `kube-apiserver-ftze3os7wcrq4jxihmvmlopctynrmhs4d6tuexttaqzwfe4ltasa` (67 characters total, 52 for the hash.

Reply:
It's mostly just nicer to deal with shorter strings. Your solution works for me. You could call base32.NewEncoding() instead of using ToLower every time, that would make it more obvious that it's not obscuring the difference between upper and lowercase letters.

Reply:
sgtm, will update shortly

Reply:
Updated to use base32 encoding and only the first 16 bytes of the hash. I ended up leaving the `strings.ToLower` in there to avoid creating two copies of the encoder string.

Reply:
Actually inclined to stick with hex encoding with `hash[:16]`. It's only a few more characters but it's just more consistent with  everything else that uses hex encoding in our codebase.

Reply:
I don't think we use hex in object names?

Reply:
base 16 is just extremely not dense.